File: analyzer.py
import stringdata
import time

def linear_search(container, element) -> int:
    print("Begin linear search for " + element)
    for i in range(len(container)):
        if container[i] == element:
            return i

    return -1

def binary_search(container, element) -> int:
    print("Begin binary search for " + element)
    low = 0
    high = len(container) - 1

    while high >= low:
        mid = (high + low) // 2
        if container[mid] < element:
            low = mid + 1
        elif container[mid] > element:
            high = mid - 1
        else:
            return mid

    return -1

def main():
    time_stamp = time.time()
    result = linear_search(stringdata.get_data(), "not_here")
    print("Result:" + str(result))
    print("Time:" + str(time.time() - time_stamp))

    time_stamp = time.time()
    result = binary_search(stringdata.get_data(), "not_here")
    print("Result:" + str(result))
    print("Time:" + str(time.time() - time_stamp))

    time_stamp = time.time()
    result = linear_search(stringdata.get_data(), "mzzzz")
    print("Result:" + str(result))
    print("Time:" + str(time.time() - time_stamp))

    time_stamp = time.time()
    result = binary_search(stringdata.get_data(), "mzzzz")
    print("Result:" + str(result))
    print("Time:" + str(time.time() - time_stamp))

    time_stamp = time.time()
    result = linear_search(stringdata.get_data(), "aaaaa")
    print("Result:" + str(result))
    print("Time:" + str(time.time() - time_stamp))

    time_stamp = time.time()
    result = binary_search(stringdata.get_data(), "aaaaa")
    print("Result:" + str(result))
    print("Time:" + str(time.time() - time_stamp))

if __name__ == '__main__':
    main()

File: analyzer.cpp
#include <iostream>
#include <vector>
#include <string>
#include "StringData.h"

// Commit #2

using namespace std;

int linear_search(const vector<string>& container, const string& element) {
    cout << "Begin linear search for " << element << endl;
    for (int i = 0; i < container.size(); i++) {
        if (container[i] == element) {
            return i;
        }
    }
    return -1;
}

int binary_search(const vector<string>& container, const string& element) {
    cout << "Begin binary search for " << element << endl;
    int low = 0;
    int high = container.size() - 1;

    while (high >= low) {
        int mid = (high + low) / 2;
        if (container[mid] < element) {
            low = mid + 1;
        } else if (container[mid] > element) {
            high = mid - 1;
        } else {
            return mid;
        }
    }

    return -1;
}

int main() {
    vector<string>& stringData = getStringData();

    long long time_stamp;
    int result;

    time_stamp = systemTimeNanoseconds();
    result = linear_search(stringData, "not_here");
    cout << "Result: " << result << endl;
    cout << "Time: " << systemTimeNanoseconds() - time_stamp << "ms" << endl;

    time_stamp = systemTimeNanoseconds();
    result = binary_search(stringData, "not_here");
    cout << "Result: " << result << endl;
    cout << "Time: " << systemTimeNanoseconds() - time_stamp << "ms" << endl;

    time_stamp = systemTimeNanoseconds();
    result = linear_search(stringData, "mzzzz");
    cout << "Result: " << result << endl;
    cout << "Time: " << systemTimeNanoseconds() - time_stamp << "ms" << endl;

    time_stamp = systemTimeNanoseconds();
    result = binary_search(stringData, "mzzzz");
    cout << "Result: " << result << endl;
    cout << "Time: " << systemTimeNanoseconds() - time_stamp << "ms" << endl;

    time_stamp = systemTimeNanoseconds();
    result = linear_search(stringData, "aaaaa");
    cout << "Result: " << result << endl;
    cout << "Time: " << systemTimeNanoseconds() - time_stamp << "ms" << endl;

    time_stamp = systemTimeNanoseconds();
    result = binary_search(stringData, "aaaaa");
    cout << "Result: " << result << endl;
    cout << "Time: " << systemTimeNanoseconds() - time_stamp << "ms" << endl;

    return 0;
}